### ISSUE TITLE
OsmAnd-iOS v4.1.1: Import of KML/KMZ Tracks: FAIL

### ISSUE THREAD

**Issue by drtbyk** [osmandapp/OsmAnd-iOS#1728]
**DESCRIPTION**
OsmAnd-iOS v4.1.1 fails to import a KML Track (see attached file). The app responds as though the Track was Imported and creates an instance in the Tracks Import Folder. But, this "track" has no track points.

[GSE22_(Africa)_-_Day_36_-_Mon,_Oct_10,_2022.zip](https://github.com/osmandapp/OsmAnd-iOS/files/7841284/GSE22_.Africa._-_Day_36_-_Mon._Oct_10._2022.zip)

**ENVIRONMENT**
iOS v15.2
iPhone Model: A2484

iPadOS v15.2
iPad Model: A2379

**Comment by dmpr0** [comment]
Hello!
OsmAnd supports import of KML/KMZ files partially.

**Comment by drtbyk** [comment]
Hello Dmitriy, What does _partially_ mean within the context of OsmAnd-iOS v4.1.1? 

Usually it means that some part of the action works to provide value including a clear description of what part of the action (Import) isn't available.  In the case of the import described above, there is no value in what the import of the KML file has done. As for converting the KML to GPX that can certainly be done - although certain data is lost in the conversion: Track Color, Track Width. 

I believe at this point I would counter that KML Import is not working.

**Comment by vshcherb** [comment]
> As for converting the KML to GPX that can certainly be done - although certain data is lost in the conversion: Track Color, Track Width.

Yes, these parts are not supported, you probably can open separate issues for each feature in that case issue will be fixed step by step. Otherwise it will take a long time to claim this issue is done

Side note: OsmAnd doesn't aim to support full functionality of KML, though it's going to support most of the features of GPX with extensions. Track width is not a standard GPX feature for example.
